### PR TITLE
chore(build):Update Dockerfiles to use redhat ubi as base image

### LINKED ISSUE #1201: RedHat UBI as the Base Image for All Containerized Applications
## Feature Request

I propose that we adopt [RedHat Universal Base Image](https://www.redhat.com/en/blog/introducing-red-hat-universal-base-image) (UBI) as our standard base image for all containerized applications  (KubeArmor, KubeArmor-Init, KubeArmor Controller, and KubeArmor Relay) going forward. We've already built and get certified the Redhat UBI based Container Images for KubeArmor and KubeArmor-Init. 

UBI, a minimal, but comprehensive, operating system image designed to be used as a base image for building container images that are portable and **certified to work with Red Hat software**. Adopting UBI as base image brings advantages:

- `Security`: UBI inherits most features of RHEL and follows updates lifecycle for the packages same as RHEL.
- `Support`: Red Hat provides support for UBI and a large community to get help with any issues. 
- `Licensing`: UBI ensure compliance with Red Hat's open source licensing requirements.

for more information please refer 
[what is redhat ubi](https://developers.redhat.com/blog/2019/10/09/what-is-red-hat-universal-base-image) and
[red-hat-developer-ubi_ebook.pdf](https://github.com/kubearmor/KubeArmor/files/11259942/red-hat-developer-ubi_ebook.pdf)

```[tasklist]
### Tasks
- [x] Identify requirements (packages and capabilities) for each application.
- [x] Build and Test UBI based Container Images (except kubearmor) on Docker, CRI-O and Containerd
- [x] Build Container Image to be run as non-root user. (It's one of the requirements to get certified by redhat)
- [x] Update CI workflow to Build and Push UBI based images
- [x] Get Container Images Certified by RedHat.
- [ ] https://github.com/kubearmor/kubearmor-relay-server/pull/35
- [ ] https://github.com/kubearmor/KubeArmor/pull/1348
- [x] Add CI workflow to test KubeArmor UBI image to be tested on bpflsm-crio env
```

### PR DESCRIPTION
**Pull request by rksharma95**
**Purpose of PR?**:

This PR updates the existing Dockerfiles to use [RedHat UBI](https://www.redhat.com/en/blog/introducing-red-hat-universal-base-image) as base image to build the `kubearmor` and `kubearmor-init` container images. RedHat UBI images inherits its foundation from RHEL ,it provides all the advantages associated with RHEL as well as eligible for Red Hat OpenShift certification.

Fixes #1201 

**Does this PR introduce a breaking change?**
No
**If the changes in this PR are manually verified, list down the scenarios covered:**:

**Additional information for reviewer?** :
_Mention if this PR is part of any design or a continuation of previous PRs_

**Checklist:**
- [x] Fixes #1201 
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change **(Not Applicable)**
- [x] This change requires a documentation update
- [x] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [x] Commit has unit tests **(Not Applicable)**
- [x] Commit has integration tests **(Added CI workflow to test build images)**

### CODE REVIEW

**Comment by rksharma95** [comment]
currently we're facing some issue with bpflsm enforcer, some of the tests are skipped until we validate and fixed the issue #1385

File: KubeArmor/BPF/shared.h [hunk 1]
```diff
@@ -219,6 +219,7 @@ static bool is_owner(struct file *file_p) {
 static bool is_owner_path(struct dentry *dent) {
   kuid_t owner = BPF_CORE_READ(dent, d_inode, i_uid);
   unsigned int z = bpf_get_current_uid_gid();
+  bpf_printk("owner %d uid %d", owner.val, z);
```
Comment (achrefbensaad) [inline KubeArmor/BPF/shared.h]: Can you remove this, it seams like a debugging message

File: KubeArmor/main.go [hunk 2]
```diff
@@ -31,8 +31,10 @@ func init() {
```
Comment (achrefbensaad) [inline KubeArmor/main.go:36]: Please resolve lint issues

**Review by achrefbensaad — CHANGES_REQUESTED** [review]
Few changes

File: KubeArmor/main.go (hunk 2 above)
Comment (achrefbensaad) [reply KubeArmor/main.go:36]: Also, please squash the commits

**Review by achrefbensaad — APPROVED** [review]
LGTM

File: KubeArmor/main.go [hunk 3]
```diff
@@ -31,8 +36,10 @@ func init() {
 
 func main() {
 	if os.Geteuid() != 0 {
-		kg.Printf("Need to have root privileges to run %s\n", os.Args[0])
-		return
+		if os.Getenv("KUBEARMOR_UBI") == "" {
+			kg.Printf("Need to have root privileges to run %s\n", os.Args[0])
+			return
+		}
```
Comment (daemon1024) [inline KubeArmor/main.go:42]: We should not do this, KubeArmor should work with just caps and not root privileged in any image or scenario.

Comment (daemon1024) [reply KubeArmor/main.go:42]: We should move to caps and non root user in standard kubearmor image as well

**Review by daemon1024 — APPROVED** [review]
LGTM.

Let's create a new issue for the inline comment.

File: KubeArmor/main.go (hunk 3 above)
Comment (rksharma95) [reply KubeArmor/main.go:42]: can we create a separate issue for it and handle there?

Comment (rksharma95) [reply KubeArmor/main.go:42]: Ref #1401

**Pull request merged**